filebeat/module/system/syslog/test/darwin-syslog-sample.log-expected.json

```diff
@@ -6,7 +6,7 @@
         "event.timezone": "-02:00",
         "fileset.name": "syslog",
         "host.hostname": "a-mac-with-esc-key",
-        "input.type": "log",
+        "input.type": "system-logs",
```

Comment:
Should this actually have changed? The `system-logs` input doesn't do data collection, shouldn't this be set to the one of logs or journald that was effective when the data was collected?

Also doesn't not knowing which of the two inputs actually did the data collection make this significantly harder to test?

Reply:
> Should this actually have changed?

Reply:
>  Should this actually have changed? 

That's how Beats works, the `input.type` is read from the config before the actual input is initialised, so on the Beat side it will require a refactoring in the whole input/publishing pipeline initialisation.

However, reading it now gave me an idea, I can set it in the ingest pipeline. I'll create a PR today for that.

Reply:
Doing it in the ingest pipeline would mean you have to do that update for every module and/or agent integration that uses this input which I don't love.

Is there a way to override this behavior? Or just have the proxy input add a processor at the end of the pipeline that overrides the field?

Reply:
Done: https://github.com/elastic/beats/pull/41246

Reply:
> Is there a way to override this behavior? Or just have the proxy input add a processor at the end of the pipeline that overrides the field?

I tried that and it did not work, I believe this field gets added after the processors are run.

> Doing it in the ingest pipeline would mean you have to do that update for every module and/or agent integration that uses this input which I don't love.

I agree it's not ideal, but to meet the FF it looked like a good compromise, at the moment this input is only used by the system integration and it's not publicly documented (it's for internal use only), so the ingest pipeline covers all cases of correct use of the input.

I'll create an GH issue to track it.

Reply:
GH issue: https://github.com/elastic/beats/issues/41258